lib/Dialect/FIRRTL/Transforms/PrefixModules.cpp

```diff
@@ -173,6 +176,23 @@ void PrefixModulesPass::renameModuleBody(std::string prefix, FModuleOp module) {
   // on the instance.
   removeDeadAnnotations(thisMod, module);
 
+  mlir::AttrTypeReplacer replacer;
+  replacer.addReplacement(
+      [&](hw::InnerRefAttr innerRef) -> std::pair<Attribute, WalkResult> {
+        StringAttr moduleName = innerRef.getModule();
+        StringAttr symName = innerRef.getName();
+
+        StringAttr newTarget;
+        if (moduleName == oldName) {
+          newTarget = module.getNameAttr();
+        } else {
+          auto target = instanceGraph->lookup(moduleName)->getModule();
+          newTarget = StringAttr::get(context, prefix + getPrefix(target) +
```

Comment:
I am not familiar with this pass, but I wonder how/why this would be the right way to compute this-- the prefix for the module being renamed does not seem like it should apply to the targets of the inner refs within?

(And this seems like it would double-count the `getPrefix()` portion when applying multiple prefixes to a module?)

Not sure we know the prefix for the module used in the inner ref here, might need to walk after prefixing is done or something?

Reply:
The graph is visited in pre-order, so the prefix of referenced modules has not yet been adjusted.

Reply:
so how is do you get the correct name for modules you haven't visited yet?